storm-core/src/jvm/org/apache/storm/task/TopologyContext.java

```diff
@@ -386,4 +388,28 @@ public ReducedMetric registerMetric(String name, IReducer reducer, int timeBucke
     public CombinedMetric registerMetric(String name, ICombiner combiner, int timeBucketSizeInSecs) {
         return registerMetric(name, new CombinedMetric(combiner), timeBucketSizeInSecs);
     }
+
+    public Timer registerTimer(String name){
+        return StormMetricRegistry.registtry().timer(metricName(name));
+    }
+
+    public Histogram registerHistogram(String name){
+        return StormMetricRegistry.registtry().histogram(metricName(name));
+    }
+
+    public Meter registerMeter(String name){
+        return StormMetricRegistry.registtry().meter(metricName(name));
+    }
+
+    public Counter registerCounter(String name){
+        return StormMetricRegistry.registtry().counter(metricName(name));
+    }
+
+    public Gauge registerGauge(String name, Gauge gauge){
+        return StormMetricRegistry.registtry().register(metricName(name), gauge);
+    }
+
+    private String metricName(String name){
+        return String.format("storm.topology.%s.%s.%s-%s", getStormId(), getThisComponentId(), getThisWorkerPort(), name);
```

Comment:
Naming of metrics appears to be spread out all over the palace.  It would be nice to have it in a single place and have a way to parse things back out of it too.